kotlin/internal/jvm/compile.bzl

```diff
@@ -280,17 +281,30 @@ def _run_kt_builder_action(ctx, rule_kind, toolchains, dirs, srcs, friend, compi
     # Collect and prepare plugin descriptor for the worker.
     args.add_all(
         "--processors",
-        plugins,
+        annotation_processors,
         map_each = _plugin_mappers.kt_plugin_to_processor,
         omit_if_empty = True,
     )
     args.add_all(
         "--processorpath",
-        plugins,
+        annotation_processors,
         map_each = _plugin_mappers.kt_plugin_to_processorpath,
         omit_if_empty = True,
     )
 
+    args.add_all(
```

Comment:
Splitting the plugin path and args gets dicey -- a plugin can have more than one argument, making the indexed based correlation hard to work with.

Reply:
Why do you want to keep track of which plugin path is for which plugin (or plugin options)? If I understand the way the compiler runs plugins correctly, there is only one plugin path, and the compiler discovers plugins that are in it.

Reply:
Good point -- got stuck in the idea that the builder would be doing the command line assembly.